bugfixtemplates/check-file-deletion-result/CheckFileDeletionResult.java

```diff
@@ -0,0 +1,45 @@
+package bugfixtemplates;
+
+import java.io.File;
+
+class CheckFileDeletionResult {
+
+    void positive1(File file) {
+        if (file.exists()) {
+            file.delete();
+        }
+    }
+
+    void negativeIf(File file) {
+        if (file.exists()) {
+            if (!file.delete()) {
+                throw new RuntimeException("Cannot write uploaded file to disk!");
+            }
+        }
+    }
+
+    void positive2(File file) {
+        if (file.exists()) {
+            file.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.